elyra/pipeline/handlers.py

```diff
@@ -86,10 +88,18 @@ async def post(self, *args, **kwargs):
         pipeline_definition = self.get_json_body()
         self.log.debug("JSON payload: %s", pipeline_definition)
 
-        pipeline = PipelineParser().parse(pipeline_definition)
+        root_dir = self.settings['server_root_dir']
+        expanded_path = get_expanded_path(root_dir)
+        validation_response = PipelineValidationManager().validate(pipeline=pipeline_definition,
+                                                                   root_dir=expanded_path)
+
+        validation_msg = validation_response.to_json()
+        json_msg = json.dumps(validation_response.to_json())
```

Comment:
We shouldn't convert to JSON twice.  Instead, I would recommend changing `validation_msg` to `response`, then perform the `to_json()` call unconditionally _after_ the validation check below (and per the next comment).